### ISSUE TITLE
lr Cann't update correctly

### ISSUE THREAD

**Issue by XueBaolu** [alibaba/FederatedScope#580]
excuse me, I defined scheduler StepLR with step_size=50. But I found the learning rate changing every 50 batch rather than 50 epoch.

And by reading codes, I find that it will redefine optimizer and scheduler in the found  _hook_on_fit_init(self, ctx) of the file federatedscope/core/trainers/torch_trainer.py.

It means, if the cfg.train.optimizer.lr is set to 0.1, a client will train with learning rate=0.1 once beginning to train its local model in a new round rather than the adjusted lr. 

Perhaps I misunderstood. if so, please pick your brain.Can you help me?

**Comment by XueBaolu** [comment]
Thank you for responding.
But I think the update of lr should be synchronized among clients. So may It can be realized if the server changes the cfg.optimizer.lr during the training once the round meets scheduler's step_size.
Can you give me some suggestions to coding?

**Comment by XueBaolu** [comment]
Sorry, I realize that it should give the cfg when creating a client, So it is not useful to change cfg.optimizer.lr during the FL process. And does it work if add an item cur_lr into the message, which is sent to clients from server at the beginning of round, to instruct clients to update its learning rate?

**Comment by XueBaolu** [comment]
OK, I will give it a try. Thank you so much.

**Comment by XueBaolu** [comment]
Wow, I can't help sharing my idea for realizing the update and synchronization of lr easily.
Because it is convenient to update lr locally for client once the self.state is given to its trainer. So I firstly add a parameter and pass the client's self.state to function trainer.train(). Next, define a variable ctx.cur_round for class Context and update it as self.state when the trainer.train() is called. Finally, run scheduler.step() for ctx.cur_round times after the redefinition of optimizer.
But please be advised that the unit of the step_size of scheduler is round, ignoring the clients' local_update_seps.
Hope it can be helpful:)
